pkg/printers/humanreadable.go

```diff
@@ -95,26 +104,90 @@ func (h *HumanReadablePrinter) EnsurePrintHeaders() {
 // See validatePrintHandlerFunc for required method signature.
 func (h *HumanReadablePrinter) Handler(columns, columnsWithWide []string, printFunc interface{}) error {
```

Comment:
Since we're in here rewriting anyway, why not fix the signature to register to use an instance of the type and the printer can be `func(runtime.Object) (whatever)` and the printers can hard cast.